### ISSUE TITLE
Support for keyboard input

### ISSUE THREAD

**Issue by georgecrawford** [watchexec/watchexec#67]
This is a nice-to-have rather than an integral requirement for the tool, but I would be delighted if `watchexec` would incorporate the keyboard command listeners from tools like `entr`:

```
entr listens for keyboard input and responds to the following commands:

     SPACE       Execute the utility immediately.  If the `-r' option is set
                 this will terminate and restart the child process as if a
                 file change event had occurred.

     q           Quit; equivalent pressing Ctrl-C.
```

or `nodemon`:
```
Whilst nodemon is running, if you need to manually restart your application, 
instead of stopping and restart nodemon, you can simply type rs with a 
carriage return, and nodemon will restart your process.
```

This would take the developer experience to the next level!

Thanks again for a fantastic tool - I evaluated about 10 different file watch utilities, and this has the best combination of API, performance and functionality I've found.

**Comment by mattgreen** [comment]
Thanks for the kind words.

I like this idea, but currently the way it handles async is pretty ad-hoc at the moment. I'd like to move to Futures so I can easily wait on several sources of input easily, but that might not be until 2.0.

**Comment by passcod** [comment]
Thinking about this today: there should also be a key to **pause** running the command on changes. Unsure yet if it should be unpaused on the same key only or unpaused also by whatever key does a manual run.

**Comment by mitranim** [comment]
Chiming in with some notes. There are two ways to go about it. One is simply to read stdin and look for specific sequences such as "h", "help", "r", "restart", and so on. This requires the user to press Enter after the characters, because terminals by default operate in line-by-line mode.

The fancier but trickier approach is to modify terminal behavior (via the termios API or equivalent). We can disable line buffering, which would allow, for example, to react to just "h" or "r" without Enter. Terminals also specially handle some key combinations, converting some of them to signals. We can disable that too; instead of signals, we'll receive [control codes](https://en.wikipedia.org/wiki/ASCII#Table_of_codes) from stdin and can interpret them liberally. For example, Ctrl+R sends the byte 18 ("device control 2" in the linked table), which we can interpret as a restart command. This approach is used by `gow`, which is a watcher somewhat similar to `watchexec` but specialized for Go. Just figured I'd share the ideas. The desire for hotkeys, and their lack in other watchers, is the reason I wrote `gow`. Supporting that in `watchexec` would be great.
